lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -136,9 +185,26 @@ public void submitTask(final String jobId, final String taskId, final URI jobJar
 
     LOG.log(Level.INFO, "Evaluator task command: " + command);
 
+    String portMappings = "";
+    Iterator<Integer> iterator = this.portProvider.iterator();
+    while (iterator.hasNext()) {
+      Integer port = iterator.next();
+      System.out.println("iter port is " + port);
+      portMappings += String.format("-p %d:%d ", port, port);
+    }
+
+    TaskContainerSettings containerSettings = null;
+    if (this.containerRegistry != null) {
```

Comment:
>containerRegistry [](start = 13, length = 17)

What if there is no ContainerRegistry. I don't believe this is a requirement to use containers. The user could use a default registry.

Reply:
If there is no containerRegistry, then it is not used. The behavior defaults to the case where the pool is not configured as a container-based pool. I tested this code for both scenarios on linux.

---
In reply to: [196514997](https://github.com/sharathmalladi/reef/pull/5#discussion_r196514997) [](ancestors = 196514997)